docs/best-practices/log-container-sidecarset.md

```diff
@@ -0,0 +1,285 @@
+      name: filebeat-config
+    - name: log
+      emptyDir: {}
+```
+
+### Automatic Injection Of FileBeat Sidecar Container
+Nginx Deployment, only contains nginx container configuration, as follows:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: nginx
+  name: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+        # need injection filebeat sidecar label
+        kruise.io/inject-filebeat: "true"
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:latest
+        volumeMounts:
+        # Share log directory with filebeat sidecar container via volumeMount
+        - mountPath: /var/log/nginx
+          name: log
+      volumes:
+      - name: log
+        emptyDir: {}
+```
+**After applying the nginx deployment to the k8s cluster, it was found that the created Pods were injected with the filebeat sidecar container**, as follows:
+```shell
+helm-charts% kubectl get pods nginx-5674976569-zdr7l -o yaml
+status:
+  containerStatuses:
+  - containerID: containerd://5330c2b32262de83ed387e5a932f61acc52e3896ddfcb22d626c43d82638faf3
+    image: docker.elastic.co/beats/filebeat:7.16.2
+    name: filebeat
+    state:
+      running:
+        startedAt: "2022-03-02T12:17:15Z"
+  - containerID: containerd://1ad335f39c134f7a66a0370a275dd95f67f5fd3d3f1fe523c955408b14887229
+    image: docker.io/library/nginx:latest
+    name: nginx
+    state:
+      running:
+        startedAt: "2022-03-02T12:17:16Z"
+```
+
+### Upgrade FileBeat Sidecar Container Independently (Version 7.16.2 -> 7.16.3)
+Below are two windows, and on the right is a client request to access the nginx service. At this point, after changing the image in filebeat sidecarSet from 7.16.2 to 7.16.3,
+**we find that the Pod is not rebuilt. And the nginx service is not interrupted during the completion of filebeat sidecar container image upgrade 7.16.3 (the nginx service has only one Pod instance)**, as follows:
+
+![k8s log sidecar](/img/docs/best-practices/update-sidecarset.gif)
+
+This feature relies on the ability of [Kruise InPlace Update](https://openkruise.io/docs/next/core-concepts/inplace-update/).
+However, this feature also has some limitations, if the sidecar container upgrade process fails, it will lead to Pod Not Ready and thus affect the business, so SidecarSet itself provides a very rich grayscale release capability.
+Refer to [Kruise SidecarSet](https://openkruise.io/zh/docs/next/user-manuals/sidecarset#sidecar%E6%9B%B4%E6%96%B0%E7%AD%96%E7%95%A5), as follows:
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: SidecarSet
+metadata:
+  name: sidecarset
+spec:
+  # ...
+  updateStrategy:
+    type: RollingUpdate
+    # Maximum unavailable quantity
+    maxUnavailable: 20%
+    # Release in batches
+    partition: 90
+    # Canary release, via pod labels
+    selector:
+      matchLabels:
+        xxxx: xxxxxx
```

Comment:
we can demonstrate that eligible labels  for selector such as (app, environment)